Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -12,311 +12,811 @@
+  ///
+  /// `AnnotatedSourceLine` encapsulates a line of source code alongside any associated diagnostics (errors, notes, etc.)
+  /// or annotations for enhanced code readability and insight. The diagnostics are represented by the `FlattenDiagnostic`
+  /// enum, which generalizes the handling of diagnostic messages and informational notes, providing a unified
+  /// approach to represent and process various types of diagnostic information.
+  ///
+  /// - Note: This struct is designed for internal use by the `DiagnosticsFormatter` to organize and manage
+  ///         source lines with their related diagnostics and annotations, facilitating a clearer understanding
+  ///         of code issues and contextual notes.
+  private struct AnnotatedSourceLine {
+    /// A collection of diagnostics associated with this line of source code.
+    ///
+    /// The diagnostics, encapsulated by instances of `FlattenDiagnostic`, offer insights into issues
+    /// identified at this specific location in the source. This enables quick identification and
+    /// resolution of problems, with `FlattenDiagnostic` providing a common interface for accessing
+    /// message text, severity, and location of diagnostics or informational notes.
+    let diagnostics: [FlattenDiagnostic]
+
+    /// The content of the source line.
+    ///
+    /// This string contains the actual code from the source file at the specified line number, serving
+    /// as a reference point for diagnostics and annotations.
+    let sourceString: String
+
+    /// The line number in the source file corresponding to `sourceString`.
+    ///
+    /// This value facilitates easy navigation to and within the source file, providing a clear reference
+    /// to where the source line is located within its broader context.
+    let sourceLineNumber: Int
+
+    /// Additional text appended to the source line for extra context or information.
+    ///
+    /// This text can include annotations, explanations, or excerpts from relevant macro expansions,
+    /// aimed at providing further insight or clarification.
+    let suffixText: String
+
+    /// Determines if the source line lacks any associated diagnostics or annotations.
+    ///
+    /// This property returns `true` when there are no diagnostics related to the line and when the
+    /// `suffixText` is empty, indicating that the line is clean or unannotated. It offers a quick
+    /// way to assess the line's status in relation to issues or annotations.
+    var isFreeOfAnnotations: Bool {
+      return diagnostics.isEmpty && suffixText.isEmpty
     }
-    return annotatedSource
+  }
+}
+
+// Extension providing all necessary APIs to maintain backward compatibility. In the future,
+// upon committing to the new API of `DiagnosticsFormatter` and diagnostic decorators,
+// all members will be marked as deprecated.
+extension DiagnosticsFormatter {
+  public var contextSize: Int {
```

Comment:
I think we should deprecate this and make `contextRange` public instead. 

```suggestion
  public var contextSize: Int {
```

Reply:
I don’t think you deprecated this one.

Also, I just remembered that we should add release notes for the deprecations.

Reply:
So if we decide to discard the idea with `full` context we would need to deprecate this property. In this case, contextSize can still be an Int. So I guess we waiting for Doug's respond here: https://github.com/apple/swift-syntax/pull/2214#discussion_r1495229284